### PR TITLE
Fix #73, add support for fetching remote theme

### LINKED ISSUE #73: Allow specifying custom themes from any source
Themes are not all that dangerous to load in, they are simply `JSON` files with some configuration to change how presentations look, we should allow users to specify custom themes with `https://` and load those from the internet.

I.e.
```
---
theme: https://github.com/maaslalani/slides/raw/main/styles/theme.json
---

# Slides

Content
```

This way the user doesn't need to transport their theme along with their slides and can even `curl` their slides already so they don't need to keep any files on disk to present.

### PR DESCRIPTION
**Pull request by suzaku**
Fix #73, add support for fetching remote theme

### Changes Introduced
- When a theme starts with "http://" or "https://", try fetch it from the internet.

### CODE REVIEW

**Comment by maaslalani** [comment]
Hey! Thanks so much for taking this on! I'll have a look soon, but from first glance it looks great

File: styles/styles.go [hunk 1]
```diff
@@ -59,20 +61,39 @@ func SelectTheme(theme string) glamour.TermRendererOption {
...
+		if strings.HasPrefix(theme, "http://") || strings.HasPrefix(theme, "https://") {
+			var resp *http.Response
+			resp, err = http.Get(theme)
+			if err != nil {
+				return getDefaultTheme()
+			}
+			defer resp.Body.Close()
+			themeReader = resp.Body
+		} else {
+			themeReader, err = os.Open(theme)
```
Comment (suzaku) [inline styles/styles.go:75]: I forgot to close the file here.

File: examples/custom_remote_theme.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+theme: https://github.com/maaslalani/slides/raw/main/styles/theme.json
+---
+
+# Slides
+
+The above title should be orange and be prefixed with `CUSTOM`.
```
Comment (suzaku) [inline examples/custom_remote_theme.md:7]: The description should be updated.

**Comment by suzaku** [comment]
I guess the next step for this PR is to store the downloaded theme in a cache directory.

File: styles/styles.go [hunk 3]
```diff
@@ -59,20 +61,41 @@ func SelectTheme(theme string) glamour.TermRendererOption {
 	case "notty":
 		return glamour.WithStyles(glamour.NoTTYStyleConfig)
 	default:
-		bytes, err := os.ReadFile(theme)
+		var themeReader io.Reader
+		var err error
+		if strings.HasPrefix(theme, "http://") || strings.HasPrefix(theme, "https://") {
```
Comment (maaslalani) [inline styles/styles.go:66]: I think this is good enough

```suggestion
		if strings.HasPrefix(theme, "http") {
```

**Review by maaslalani — APPROVED** [review]
Tried this out, works great!

File: styles/styles.go (hunk 3 above)
Comment (suzaku) [reply styles/styles.go]: What if it's something like `httpie.xxx`?

**Pull request merged**